pulsar/internal/connection.go

```diff
@@ -744,6 +745,50 @@ func (c *connection) handleAuthChallenge(authChallenge *pb.CommandAuthChallenge)
 	c.writeCommand(baseCommand(pb.BaseCommand_AUTH_RESPONSE, cmdAuthResponse))
 }
 
+func (c *connection) handleSendError(sendError *pb.CommandSendError, cmdError *pb.CommandError) {
+	c.log.Warnf("Received send error from server: [%v] : [%s]", sendError.GetError(), sendError.GetMessage())
+
+	requestID := cmdError.GetRequestId()
+	producerID := sendError.GetProducerId()
+
+SendError:
+	switch *sendError.Error {
+	case pb.ServerError_NotAllowedError:
+		c.pendingLock.Lock()
+		request, ok := c.pendingReqs[requestID]
+		if !ok {
+			c.log.Warnf("Received unexpected error response for request %d of type %s",
+				requestID, cmdError.GetError())
+			c.pendingLock.Unlock()
+			return
+		}
+
+		delete(c.pendingReqs, requestID)
+		c.pendingLock.Unlock()
+
+		errMsg := fmt.Sprintf("server error: %s: %s", cmdError.GetError(), cmdError.GetMessage())
+		request.callback(nil, errors.New(errMsg))
+		break SendError
+	case pb.ServerError_TopicTerminatedError:
+		c.listenersLock.RLock()
+		producer, ok := c.listeners[producerID]
+		c.listenersLock.RUnlock()
+
+		if ok {
+			producer.ConnectionClosed()
+		} else {
+			c.log.
```

Comment:
wrong indentation?

Reply:
No, this is automatically formatted by golanglint, probably for aesthetic reasons